CSC8503CoreClasses/AnimationObject.h

```diff
@@ -0,0 +1,71 @@
+#include "GameObject.h"
+#include "Transform.h"
+#include "CollisionVolume.h"
+#include "MeshAnimation.h"
+#include "../NCLCoreClasses/Camera.h"
+#include "../OpenGLRendering/OGLRenderer.h"
+
+
+
+using std::vector;
+class NetworkObject;
+class RenderObject;
+class PhysicsObject;
+class MeshAnimation;
+class MeshMaterial;
+namespace NCL::CSC8503 {
+	
+	class AnimationObject : public GameObject {
```

Comment:
This all looks good and will work, but would it not make more sense to have AnimationObject as a component of GameObject, rather than a sub-class of it? In the same way that RenderObject, NetworkObject, and CollisionObject are all components.